consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -2,16 +2,30 @@ package io.buoyant.consul
 
 import com.twitter.finagle._
 import com.twitter.util._
-import io.buoyant.consul.v1.{UnexpectedResponse, ServiceNode}
+import io.buoyant.consul.v1.{ServiceNode, UnexpectedResponse}
+import io.buoyant.namer.Metadata
 
 class CatalogNamer(
   idPrefix: Path,
-  api: v1.CatalogApi,
-  includeTag: Boolean = false
+  catalogApi: v1.CatalogApi,
+  agentApi: v1.AgentApi,
+  includeTag: Boolean = false,
+  setHost: Boolean = false
 ) extends Namer {
 
   import CatalogNamer._
 
+  private[this] lazy val domainFuture: Future[String] =
```

Comment:
Unfortunately it doesn't support long polling. Would you like a `Var.async` with some interval? If yes, what the interval would be ok? Something like `10.seconds`?